File: Iteratia1/ExtensieRazboi/src/PlayerHand.java
import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public class PlayerHand {

   // private Deck deck;
    private int noOfCards = 52; //carti ramase in pachet
    private int initialNumber; //numarul de carti pe care le primeste jucatorul la inceput
    private List<Card> playersCards; //cartile din mana jucatorului
    private boolean isTurn; //true cand e randul jucatorului
    private Card prisoner;

    public PlayerHand(int initialNumber, Deck carti) {
        this.initialNumber = initialNumber;
       // deck=carti;
        drawInitialCards(carti);

           // System.out.println(playersCards);

    }
    public List<Card> getCards(){
        return playersCards;
    }

    public int nrCarti(){

        return playersCards.size();
    }

    public void setInitialNumber(int initialNumber) {
        this.initialNumber = initialNumber;
    }


    public boolean getTurn() {
        return isTurn;
    }

    public void setTurn() {
        isTurn = true;
    }

    /*public void makeDeck(){
        deck=new ArrayList<>();
        for(int i=2;i<=14;i++) {
            for (int j = 1; j <= 4; j++) {
                deck.add(new Card(i, j));
            }
        }
    }*/

    public void drawInitialCards(Deck deck){ //impartirea cartilor la inceput

        playersCards = new ArrayList<>();
        for(int i=0; i<initialNumber; i++) {
         try{
            playersCards.add(deck.extractOneCard());
        }catch(IndexOutOfBoundsException e){}


         /*
            Random rnd=new Random();
            int index = rnd.nextInt(noOfCards);
            try {
                playersCards.add(deck.get(index)); //adauga in mana jucatorului cartea de pe poz index din pachet
            } catch(IndexOutOfBoundsException e){}
            noOfCards--;
            try {
                deck.remove(index);
            } catch(IndexOutOfBoundsException e){}*/
        }


    }

    public void addCard(Card carte){
        playersCards.add(carte);
    }


    public Card placeCard(){  // trebuie adaptata functia in cazul in care jucatorul castiga mai multe carti, nu e random
        if(playersCards.size()>0) {
            try {
                Card carte = playersCards.get(0);
                playersCards.remove(0);

                return carte;
            }catch(IndexOutOfBoundsException e){}
        }
        return null;

       /* Random rnd=new Random();
        int index = rnd.nextInt(noOfCards);
        playersCards.add(deck.get(index)); //adauga in mana jucatorului cartea de pe poz index din pachet
        noOfCards--;
        deck.remove(index);
        */
    }


    public void putCard(int index){ //fiecare jucator are un deck diferit, trebuie revizuit
        //ar trebui sa puna o carte jos in functie de reguli
        Card chosenCard = playersCards.get(index);
        int color = chosenCard.getColor();
        int number = chosenCard.getNumber();
        //System.out.println("Number= " + number + ",  culoare= " + color);
        //apeleaza functia de reguli cu number si color ca parametri; daca valideaza ca fiind corect pune cartea jos
        //altfel trimite un mesaj ca trebuie sa alegem alta carte

        //deck.add(chosenCard);
        noOfCards++;
        playersCards.remove(index);

       // System.out.println("Number of cards: " + playersCards.size());

    }

    public void printCards(){
        System.out.println(playersCards.toString());
    }
}

File: ProjectIP/ProjectIP/PlayerHand.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ProjectIP
{
    class PlayerHand
    {
       private int initialNumber; //numarul de carti pe care le primeste Player-ul la inceput
       private List<Card> playersCards; // cartile din mana Player-ului
       private bool isTurn; //true cand e randul jucatorului
    

       public void setInitialNumber(int initialNumber) //il modificam in functie de tipul jocului
        { this.initialNumber = initialNumber; }
       public int getInitialNumber() => initialNumber;

       public bool getTurn() => isTurn;

        public int getNrCards() => playersCards.Count;

       public void setTurn() => isTurn = false; //default e false, devine true in timpul jocului

        /*  public void drawInitialCards(Deck deck){ //ar trebui pus in game
                //impartirea cartilor la inceput
                int initialNumber = getInitialNumber();

                List<Card> playersCards = new List<Card>();
                for (int i = 0; i < initialNumber; i++){
                    Random rnd = new Random();
                    List<Card> deckList = deck.getDeck();
                    int index = rnd.Next(deckList.Count);
                    playersCards.Add(deckList[index]); //adauga in mana jucatorului cartea de pe poz index din pachet

                    deckList.RemoveAt(index);
                    deck.setDeck(deckList);
                    }
                } */

        //jucatorul trebuie sa ridice o carte sau mai multe de pe masa
        //jucatorul ia toate cartile de pe masa(deck, deck.Count)
        public void drawCard(Deck deck, int nrCards) //tot in game
            {
            //deck ul de pe table ar treb sa fie tip stack
                Random rnd = new Random();
                List<Card> deckList = deck.getDeck();
                int index = rnd.Next(deckList.Count);
                playersCards.Add(deckList[index]); //adauga in mana jucatorului cartea de pe poz index din pachet
                deckList.RemoveAt(index);

                deck.setDeck(deckList);
            }

            
            // jucatorul treb sa pune jos o carte (la razboi, index=0) 
            //trebuie sa returneze si id ul jucatorului 
            public Card putCard(int index)  
            {
                //ar trebui sa puna o carte jos in functie de reguli
                Card chosenCard = playersCards[index];
               // CardColor color = chosenCard.getColor();
               // int number = chosenCard.getNumber();
               // Console.WriteLine("Number= " + number + ",  culoare= " + color);
                //apeleaza functia de reguli cu number si color ca parametri; daca valideaza ca fiind corect pune cartea jos
                //altfel trimite un mesaj ca trebuie sa alegem alta carte

               // List<Card> deckList = deck.getDeck();
                // deckList.Add(chosenCard);
              //  deck.setDeck(deckList);
                playersCards.RemoveAt(index);

            return chosenCard;
                //Console.WriteLine("Number of cards: " + playersCards.Count);

            }

            public void printCards()
            {
                Console.WriteLine(playersCards.ToString());
            }
        
    }
}
